File: Competitive Programming 3/Section 1.3.3/11586 - Train Tracks/Java/Main.java
import java.util.Scanner;

public class Main {
    static Scanner s = new Scanner(System.in);
    public static void main(String[] args){
        for(int cases = Integer.parseInt(s.nextLine()); cases > 0 ; cases --){
            String[] tk = s.nextLine().split(" ");
            int FFs=0,MMs=0;
            for(int i = 0; i < tk.length; i++){
                if(tk[i].compareTo("FF")==0){
                    FFs++;
                }else if(tk[i].compareTo("MM")==0){
                    MMs++;
                }
            }
            System.out.printf("%s\n",tk.length==1||Math.abs(FFs-MMs)>0?"NO LOOP":"LOOP");
        }
    }
}

File: Competitive Programming 3/Section 1.3.3/11586 - Train Tracks/Python/main.py
def main():
    cases = int(input())
    for _ in range(0,cases):
        FFs,MMs=0,0
        tks = input().split()
        for piece in tks:
            if piece == 'FF':
                FFs+=1
            elif piece == 'MM':
                MMs+=1
        print('NO LOOP' if len(tks)==1 or abs(FFs-MMs)>0 else 'LOOP')
main()